ge/org.osate.ge.errormodel/src/org/osate/ge/errormodel/ui/viewmodels/ErrorTypeExtendTypeModel.java

```diff
@@ -0,0 +1,151 @@
+ * Software. The parties who own the Third Party Software ("Third Party Licensors") are intended third party benefici-
+ * aries to this license with respect to the terms applicable to their Third Party Software. Third Party Software li-
+ * censes only apply to the Third Party Software and not any other portion of this program or this program as a whole.
+ */
+package org.osate.ge.errormodel.ui.viewmodels;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.eclipse.xtext.resource.IEObjectDescription;
+import org.osate.ge.BusinessObjectSelection;
+import org.osate.ge.aadl2.ui.AadlModelAccessUtil;
+import org.osate.ge.swt.BaseObservableModel;
+import org.osate.ge.swt.selectors.SingleSelectorModel;
+import org.osate.xtext.aadl2.errormodel.errorModel.ErrorModelPackage;
+import org.osate.xtext.aadl2.errormodel.errorModel.ErrorType;
+
+public class ErrorTypeExtendTypeModel extends BaseObservableModel implements SingleSelectorModel<ErrorType> {
+	private static class Option {
+		public Option(final IEObjectDescription desc) {
+			this.type = (ErrorType) desc.getEObjectOrProxy();
+			this.qualifiedName = ErrorModelViewModelUtil.getQualifiedName(desc);
+		}
+
+		public final ErrorType type;
+		public final String qualifiedName;
+	}
+
+	private BusinessObjectSelection bos;
+	private Map<Object, Option> uriToOptionMap;
+	private String error;
+
+	/**
+	 * Creates a new instance
+	 * @param bos the initial business object selection
+	 */
+	public ErrorTypeExtendTypeModel(final BusinessObjectSelection bos) {
+		setBusinessObjectSelection(bos);
+	}
+
+	/**
+	 * Refreshes the internal state of the model based on the specified business object selection
+	 */
+	public void setBusinessObjectSelection(final BusinessObjectSelection value) {
+		this.bos = Objects.requireNonNull(value, "value must not be null");
+
+		uriToOptionMap = new HashMap<>();
+		error = null;
+
+		final List<ErrorType> selectedErrorTypes = value.boStream(ErrorType.class).collect(Collectors.toList());
+		if (selectedErrorTypes.size() > 1) {
+			error = "<Multiple Error Types Selected>";
+		} else if (selectedErrorTypes.size() == 0) {
+			error = "<No Error Types Selected>";
+		} else {
+			final ErrorType selectedErrorType = selectedErrorTypes.get(0);
+			if (selectedErrorType.eResource() == null) {
+				error = "<No Resource Available>";
+			} else {
+				final Collection<IEObjectDescription> descriptions = AadlModelAccessUtil.getAllEObjectsByType(
+						selectedErrorType.eResource(), ErrorModelPackage.eINSTANCE.getErrorType());
+				final URI selectedErrorTypeURI = EcoreUtil.getURI(selectedErrorType);
+				uriToOptionMap.put(new Object(), null);
```

Comment:
Don't add a dummy object to the map. See `ErrorBehaviorModel.getElements()` for an example of how to include a null values.